platform/wm/rhodes/browser/CEBrowserEngine.cpp

```diff
@@ -1157,4 +1164,134 @@ BOOL CEBrowserEngine::ZoomTextOnTab(int nZoom, UINT iTab)
 }
 
 
+DWORD WINAPI CEBrowserEngine::NetworkWindowThread( LPVOID lpParameter )
+{
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Started";
+	CEBrowserEngine * pEng = (CEBrowserEngine*) lpParameter;
+
+	TCHAR szWindowText[MAX_PATH];
+	TCHAR szWindowClass[MAX_PATH];
+	HWND hCurrentWindow = NULL;
+
+	ZeroMemory(szWindowText,MAX_PATH);
+	ZeroMemory(szWindowClass,MAX_PATH);
+
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread";
+	LOG(INFO) + pEng->m_bNavigationComplete;
+	
+	while(!pEng->m_bNavigationComplete)
+	{
+		HWND hCurrentWindow=FindWindow(TEXT("Dialog"),TEXT("Enter Network Password"));
+		if(hCurrentWindow != NULL)
+		{
+
+#ifdef _DEBUG
+			GetWindowText(hCurrentWindow,szWindowText,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  Title";
+			LOG(INFO) + szWindowText;
+			ZeroMemory(szWindowClass,MAX_PATH);
+			GetClassName(hCurrentWindow,szWindowClass,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  class";
+			LOG(INFO) + szWindowClass;
+			LONG lIndex = GetWindowLong(hCurrentWindow,GWL_ID);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  id";
+			LOG(INFO) + lIndex;
+#endif
+			::EnumChildWindows(hCurrentWindow,(WNDENUMPROC)NetworkWndProc,0);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended (0)";
+			pEng->m_bNavigationComplete = TRUE;
+			return 0;	
+		}
+		
+	}
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended";
+	pEng->m_bNavigationComplete = TRUE;
```

Comment:
What i mean to say is that why are you setting it to true. This variable will already be true... It will reach to that line only if it exits from top while loop.